sbt-scalajs-bundler/src/main/scala/scalajsbundler/Stats.scala

```diff
@@ -107,32 +141,10 @@ object Stats {
       assets.map(a => outputPath.getOrElse(altDir).resolve(a.name).toFile)
   }
 
-  implicit val assetsReads: Reads[Asset] = (
-    (JsPath \ "name").read[String] and
-    (JsPath \ "size").read[Long] and
-    (JsPath \ "emitted").read[Boolean] and
-    (JsPath \ "chunkNames").read[List[String]]
-  )(Asset.apply _)
-
-  implicit val errorReads: Reads[WebpackError] = (
-    (JsPath \ "moduleName").read[String] and
-      (JsPath \ "message").read[String] and
-      (JsPath \ "loc").read[String]
-    )(WebpackError.apply _)
-
-  implicit val warningReads: Reads[WebpackWarning] = (
-    (JsPath \ "moduleName").read[String] and
-      (JsPath \ "message").read[String]
-    )(WebpackWarning.apply _)
-
-  implicit val statsReads: Reads[WebpackStats] = (
-    (JsPath \ "version").read[String] and
-    (JsPath \ "hash").read[String] and
-    (JsPath \ "time").read[Long] and
-    (JsPath \ "outputPath").readNullable[String].map(x => x.map(new File(_).toPath)) and // It seems webpack 2 doesn't produce outputPath
-    (JsPath \ "errors").read[List[WebpackError]] and
-    (JsPath \ "warnings").read[List[WebpackWarning]] and
-    (JsPath \ "assets").read[List[Asset]]
-  )(WebpackStats.apply _)
+  implicit val assetsReads: Reads[Asset] = Json.reads[Asset]
+  implicit val errorReads: Reads[WebpackError] = Json.reads[WebpackError]
+  implicit val warningReads: Reads[WebpackWarning] = Json.reads[WebpackWarning]
+  implicit val pathReads: Reads[Path] = Reads.StringReads.map(new File(_).toPath)
+  implicit val statsReads: Reads[WebpackStats] = Json.using[Json.WithDefaultValues].reads[WebpackStats]
```

Comment:
These changes don't seem to have anything to do with the purpose of this PR, have they?

Reply:
@sjrd It does because you need to update the JSON decoders to parse the new `Option` types that have been added. So rather than explicitly modify the individual fields it parses as Options, this solves the problem using derived decoders, which is also fewer lines and I don't think there are any downsides since it's compile-time derivation (maybe slightly longer time taken compiling?)

Reply:
Since the model has changed, `Reads` do have to be modified, and derived decoders are a simplification of those changes.